File: src/main/java/com/sisqueslabs/nexora/api/contexts/chat/domain/exceptions/ModelNotFoundException.java
package com.sisqueslabs.nexora.api.contexts.chat.domain.exceptions;

import com.sisqueslabs.nexora.api.core.domain.apperr.NotFoundException;

/**
 * The requested model doesn't exist in the catalog.
 */
public class ModelNotFoundException extends NotFoundException {

    public ModelNotFoundException(String requestedModel, String availableModel) {
        super("model \"%s\" not found, only \"%s\" is available in v0".formatted(requestedModel, availableModel));
    }
}

File: src/contexts/chat/domain/exceptions/model-not-found.exception.ts
import { BaseException } from '@sisques-labs/nestjs-kit';

/**
 * The requested model doesn't exist in the catalog.
 */
export class ModelNotFoundException extends BaseException {
  constructor(requestedModel: string, availableModel: string) {
    super(
      `model "${requestedModel}" not found, only "${availableModel}" is available in v0`,
    );
  }
}
